var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -561,26 +587,33 @@ def common_config_args(self):
             self.prefix,
             "-v",
             "-opensource",
-            "-{0}opengl".format("" if "+opengl" in spec else "no-"),
             "-{0}".format("debug" if "+debug" in spec else "release"),
             "-confirm-license",
             "-optimized-qmake",
             "-no-pch",
         ]
 
+        # Windows currently only supports the desktop provider for opengl
+        if "+opengl" in spec:
+            config_args.append("-opengl")
+            if IS_WINDOWS:
+                config_args.append("desktop")
+        else:
+            config_args.append("-noopengl")
+
         use_spack_dep = self._dep_appender_factory(config_args)
 
         if "+gui" in spec:
             use_spack_dep("freetype")
-            if not MACOS_VERSION:
+            if LINUX_VERSION:
                 config_args.append("-fontconfig")
         else:
             config_args.append("-no-freetype")
             config_args.append("-no-gui")
 
         if "+ssl" in spec:
-            pkg = spec["openssl"]
             config_args.append("-openssl-linked")
+            pkg = spec["openssl"]
```

Comment:
This specific change seems non-essential - did you think this organization was better?